simplex/docs/reconfiguration.md

```diff
@@ -0,0 +1,471 @@
+   uint64 prev_vm_block_seq = 4; // The sequence of the previous VM block
+   uint64 sealing_block_seq = 5; // The sequence number of the sealing block
+   BlockValidationDescriptor block_validation_descriptor = 6; // Describes how to validate the blocks of the next epoch
+   NextEpochApprovals next_epoch_approvals = 7; // The epoch change approvals of the next epoch by at least n-f nodes.
+}
+```
+
+- The validator set of the epoch numbered `epoch_number` is derived from `p_chain_reference_height`.
+
+- The `next_p_chain_reference_height` is the P-chain height of the next epoch, otherwise it is set to `0`.
+
+- The `prev_vm_block_seq` is the sequence number of the previous VM block, and it is used to efficiently find the last VM block upon startup.
+
+- The `sealing_block_seq` is the sequence number of the sealing block of the epoch, and it is set to `0` before the sealing block is created.
+  It is used to determine when the Simplex instance can transition to the next epoch. It is safe to transition to the next epoch once the sealing block is finalized.
+
+- The `block_validation_descriptor` describes how to validate the blocks of the next epoch.
+  It is encoded in blocks starting from the block that has `next_p_chain_reference_height > 0` until and including the sealing block.
+  It is used to verify the quorum certificates of blocks that are built in the next epoch.
+
+- The `next_epoch_approvals` is a canoto message that contains the approvals of the next epoch by at least `n-f` nodes.
+  It is required in order to create the sealing block of the epoch and indicates that the nodes comprising the next epoch's validator set
+  are ready to start the next epoch. It prevents cases where nodes in the current epoch creating a sealing block and moving to the next epoch
+  while the nodes of the next epoch still replicating blocks and cannot yet participate in the consensus protocol.
+
+- The genesis block does not contain the Simplex epoch information, and it is implicitly set to be the zero values of the fields above.
+
+
+### Auxiliary information encoding
+
+Apart from the ICM epoch information and the Simplex epoch information, the block also contains auxiliary information
+that may be used by other protocols that require their information to be collected via the process of block building by the protocol participants.
+A common way for such information collection is to have each participant supply its input as part of the auxiliary information in the block, when it is its turn to build a block.
+
+The auxiliary information is encoded as a canoto message with the following schema:
+
+```proto
+message AuxiliaryInfo {
+   bytes info = 1; // The auxiliary information. Can be empty if no auxiliary information is provided.
+   uint64 prev_aux_info_seq = 2; // The sequence number of the previous block containing auxiliary information.
+   uint32 application_id = 3; // The application ID of the auxiliary information (if applicable).
+}
```

Comment:
I don't _really_ have a strong preference here. But I do generally find version tags in proto as an antipattern (as the message structure itself is typically preferred to handle the versioning in proto).

Would it make sense for `info` to be a OneOf rather than coupling `info` and `application_id`? Or would it make sense for `info` to handle its own versioning?

Technically we don't need to actually decide on this until we have a non-zero `application_id`... But I figured it made sense to just finalize this now.